lesson_12/io/io_app/src/main/java/com/codedifferently/lesson12/factory/randycastro/CsvDataLoader.java

```diff
@@ -0,0 +1,97 @@
+package com.codedifferently.lesson12.factory.randycastro;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import com.codedifferently.lesson12.factory.LibraryCsvDataLoader;
+import com.codedifferently.lesson12.models.CheckoutModel;
+import com.codedifferently.lesson12.models.LibraryDataModel;
+import com.codedifferently.lesson12.models.LibraryGuestModel;
+import com.codedifferently.lesson12.models.MediaItemModel;
+
+
+// create a path for each file 
+@Service
+public class CsvDataLoader implements LibraryCsvDataLoader{
```

Comment:
I can see this is still in progress, so feel free to continue working it if you'd like to refine.